openmmtools/forcefactories.py

```diff
@@ -174,6 +174,240 @@ def restrain_atoms(thermodynamic_state, sampler_state, restrained_atoms, sigma=3
+
+    allowable_nb_methods = {openmm.NonbondedForce.CutoffPeriodic, openmm.NonbondedForce.CutoffNonPeriodic}
+    assert(nonbonded_force.getNonbondedMethod() in allowable_nb_methods)
+
+
+    # call constructor
+    # The 'energy_prefactor' allows us to easily change sign, or e.g. halve the energy if needed
+    new_force = openmm.CustomNonbondedForce(energy_prefactor+energy_expression)
+    new_force.addPerParticleParameter('charge')
+    new_force.addPerParticleParameter('sigma')
+    new_force.addPerParticleParameter('epsilon')
+
+
+    # go through all of the setter and getter methods
+    new_force.setCutoffDistance(nonbonded_force.getCutoffDistance())
+    #new_force.setEwaldErrorTolerance(nonbonded_force.getEwaldErrorTolerance())
+    #new_force.setForceGroup(nonbonded_force.getForceGroup())
+    new_force.setNonbondedMethod(nonbonded_force.getNonbondedMethod())
+    #new_force.setPMEParameters(*nonbonded_force.getPMEParameters())
+    #new_force.setReactionFieldDielectric(nonbonded_force.getReactionFieldDielectric())
+    #new_force.setReciprocalSpaceForceGroup(nonbonded_force.getReciprocalSpaceForceGroup())
+    new_force.setUseSwitchingFunction(nonbonded_force.getUseSwitchingFunction())
+    new_force.setSwitchingDistance(nonbonded_force.getSwitchingDistance())
+    new_force.setUseLongRangeCorrection(nonbonded_force.getUseDispersionCorrection())
+
+
+    # now add all the particle parameters
+    num_particles = nonbonded_force.getNumParticles()
+    for i in range(num_particles):
+        new_force.addParticle(nonbonded_force.getParticleParameters(i))
+
+    # now add all the exceptions? # TODO: check if we want to do this...
+    #    for exception_index in range(nonbonded_force.getNumExceptions()):
+    #        new_force.addException(nonbonded_force.getExceptionParameters(exception_index))
+
+    # TODO: There's probably a cleaner, more Pythonic way to do this
+    # (this was the most obvious way, but may not work in the future if the OpenMM API changes)
+
+    return new_force
+
+
+def split_nb_using_interaction_groups(system, md_topology):
+    """Construct a copy of system where its nonbonded force has been replaced
+    with two nonbonded forces, one using an interaction group restricted to
+    water-water interactions, and the other using interaction groups
+    restricted to water-solute and solute-solute interactions. Water-water
+    interactions are in force group 0, and all other interactions are in force
+    group 1.
+    """
+
+    # create a copy of the original system: only touch this
+    new_system = copy.deepcopy(system)
+
+    # find the default nonbonded force
+    force_index, nb_force = forces.find_forces(new_system, openmm.NonbondedForce, only_one=True)
+    # create copies for each interaction. Only half in solvent/solvent and solute/solute as we double-count.
+    nb_only_solvent_solvent = clone_nonbonded_parameters(nb_force, energy_prefactor='0.5*')
```

Comment:
Why do you want to multiple the energy by 0.5?

Reply:
Mistakenly assumed double-counting -- will revert